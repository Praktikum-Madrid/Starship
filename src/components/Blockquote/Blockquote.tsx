import React, { FC } from 'react';

interface IProps {
  text: string,
}

const styleBlockquote = {
  fontFamily: 'Roboto, sans-serif',
  fontSize: '13px',
  fontStyle: 'oblique',
  padding: '0 25px',
  margin: '15px 0',
  borderLeft: '1px solid #ddd',
  lineHeight: '150%',
};

const Blockquote: FC<IProps> = ({ text }) => <blockquote style={styleBlockquote}>{text}</blockquote>;

export default Blockquote;
