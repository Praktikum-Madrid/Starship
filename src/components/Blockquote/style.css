.blockquote {
  font-family: Roboto, sans-serif;
  font-size: 13px;
  font-style: oblique;
  color: #222;
  padding: 0 25px;
  margin: 15px 0;
  border-left: 1px solid #ddd;
  line-height: 150%;
}
