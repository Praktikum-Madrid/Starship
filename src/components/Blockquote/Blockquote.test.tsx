import React from 'react';
import renderer from 'react-test-renderer';
import Blockquote from 'components/Blockquote/Blockquote';
import {render, screen} from '@testing-library/react';


it('Снэпшот соответствует ожидаемому', () => {
  const app = renderer.create(<Blockquote text='Test'/>).toJSON();

  // Сравниваем со снэпшотом
  expect(app).toMatchSnapshot();
})

// Тест с использованием REACT TEST LIBRARY
describe('Пропсы передаются корректно', () => {
  test('renders button', () => {
    render(<Blockquote text='Test'/>);
    expect(screen.getByText('Test')).toBeInTheDocument();
  })
})
