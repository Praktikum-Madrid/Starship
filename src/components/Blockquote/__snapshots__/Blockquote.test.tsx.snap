// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Снэпшот соответствует ожидаемому 1`] = `
<blockquote
  className="blockquote"
>
  Test
</blockquote>
`;
